apollo-client/src/main/java/com/ctrip/framework/apollo/spring/annotation/ApolloAnnotationProcessor.java

```diff
@@ -244,9 +250,9 @@ private Object resolvePropertyValue(String beanName, String placeHolder) {
     return propertyValue;
   }
 
-  private Object parseJsonValue(String json, Type targetType) {
+  private Object parseJsonValue(String json, Type targetType, String datePattern) {
     try {
-      return GSON.fromJson(json, targetType);
+      return DATEPATTERN_GSON_MAP.computeIfAbsent(datePattern, (pattern) -> new GsonBuilder().setDateFormat(pattern).create()).fromJson(json, targetType);
```

Comment:
The default datePattern value of `Gson` is [null](https://github.com/google/gson/blob/51cce39e0064a755d32f820670770f82519039de/gson/src/main/java/com/google/gson/Gson.java#L267) while the the default datePattern value of `ApolloJsonValue` is empty.
I'm not sure whether they work the same way but it's better to keep the original logic if user has no intention to config the datePattern.

Reply:
:thumbsup: Yes, it's best to keep the existing logic unchanged.  I'm making the changes now. :smiley: